### ISSUE TITLE
Almost there

### ISSUE THREAD

**Issue by Antonstr** [bessman/navbot#2]
Hi Alexander

Trust that you are well

I managed to borrow a C94 EVK from the local Ublox agent.

It is not giving me the same results as my own C94 EVK that I purchased from them 6 years ago and used
for the development of my own device.

What is very positive is that the EVK gives me a RTK Float solution that is better than autonomous but not as good as a RTK Fixed
solution.

I let the vehicle navigate to one of the markers, and after just over half a turn the vehicle broke out and headed pretty much directly to the target.

After about 10 seconds it maintained its present course. The distance to the marker however did not change despite the vehicle covering just under 3 meters of the calculated 6.02 meters.

The vehicle, still being severely underpowered, then reached a very slight incline that slowed it down to about half speed and struggled to overcome the small incline. It then went into a right hand turn and broke out again after just over a full turn, this was due to the ‘track’ value being 0.00 after the vehicle lost speed.

It was more or less within 5 degrees aligned with the target judging by eye.

It never reached the target but came very close to it. I repeated the process from a number of starting positions trying to keep the path once aligned downhill as much as possible but it is not always possible.

I am trying to get hold of 2 more powerful motors and a higher, 40 Amp Hbridge motor controller.

Presently nobody seems to have stock but I am looking daily.

Your thoughts on the distance issue?

Thank you

Sincerely

Anton

**Comment by bessman** [comment]
Hi Anton,

When you say that the distance to the marker did not change, do you mean that the distance value in the "x to target" message does not update? I believe that was fixed in #1, which your subsequent test also verified, no?

Or perhaps it does update, but the value does not reflect the actual distance to the target very well? If so, does that only happen when it is close to the target, i.e. is the offset more or less constant?

**Comment by Antonstr** [comment]
Hi Alexander

It was sorted in a way, yes. The x to the target requires continuous update
as the vehicle overshoots the target.

I have set the update from 10 to 1,2,3,4,5,6,7,8,9 to 15 seconds between 4
and 6 seconds. She maintains the track very well but if it reaches the
target with time remaining it overshoots the target and does not stop but
carries on driving.

I tried placing her further and closer to the target to time it updating
when at the target.

That did not work because the vehicle is not consistent speed wise,
sometimes she breaks out in the first curve sometimes second one. All
depends on the speed of the vehicle. I will know tomorrow about better
motors, wheels and motor-controller that will allow the vehicle to maintain
speed much better.

If i physically pick it up and turn it till it "breaks out and slowly walk
to the target and the keep her still above the target she does stop after a
few seconds, I would guess about 10 to 15 seconds if I keep her still
enough and do not lean over the antennas

I suspect that if the distance is updated every second she will perform as
expected

Thank you


Sincerely

Anton

**Comment by bessman** [comment]
Right, I see. Yes, driving forward blindly for a set amount of time is a very naive way to do this and inevitably leads to this kind of problem.

The correct way to solve this issue is to use asyncio and keep track of the position in the background. I don't think I'll have time to implement that in the near future, though. For now, I would recommend setting the wait time to a small value and maybe increasing the bearing tolerance, to avoid the robot constantly adjusting its bearing.

**Comment by Antonstr** [comment]
Hi Alexander

Thank you, it is appreciated.


Sincerely

Anton
